### PR TITLE
fix for out of bounds sound resampling

### PR DESCRIPTION
**Pull request by birarda**
This fixes a windows crash on cleanup of Sound resource thanks to heap corruption

### CODE REVIEW

File: libraries/audio/src/Sound.cpp [hunk 1]
```diff
@@ -94,22 +94,32 @@ void Sound::downloadFinished(QNetworkReply* reply) {
...
-    
     if (_isStereo) {
         for (int i = 0; i < numSourceSamples; i += 4) {
-            destinationSamples[i / 2] = (sourceSamples[i] / 2) + (sourceSamples[i + 2] / 2);
-            destinationSamples[(i / 2) + 1] = (sourceSamples[i + 1] / 2) + (sourceSamples[i + 3] / 2);
+            if (i + 2 >= numSourceSamples) {
+                destinationSamples[i / 2] = sourceSamples[i];
+                destinationSamples[(i / 2) + 1] = sourceSamples[i + 1];
+            } else {
+                destinationSamples[i / 2] = (sourceSamples[i] / 2) + (sourceSamples[i + 2] / 2);
```
Comment (PhilipRosedale) [inline libraries/audio/src/Sound.cpp]: wouldn't this more correctly fix rounding errors... 

(sourceSamples[i]  + sourceSamples[i + 2]) / 2

**Comment by hifi-gustavo** [comment]
Build console output for commit: e568f47be9a909947691d2c06c3adc88f3fe494b available at: https://gist.github.com/99535563e6076def740f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4407.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4407.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 580b74050ce4bfa27ab5dc147ed88d5452951b85 available at: https://gist.github.com/99535563e6076def740f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4407.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4407.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 271a4e2cae84a95639685def8491b8af6a80df49 available at: https://gist.github.com/99535563e6076def740f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4407.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4407.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c660b771db9e301bd2985c293b4b8db531373722 available at: https://gist.github.com/99535563e6076def740f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4407.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4407.dmg

**Pull request merged**